ValveResourceFormat/IO/FileExtract.cs

```diff
@@ -1,16 +1,44 @@
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Text;
 using SkiaSharp;
 using ValveResourceFormat.ResourceTypes;
 
 namespace ValveResourceFormat.IO
 {
+    public class ContentFile
+    {
+        public byte[] Data { get; set; }
```

Comment:
> I had to undo the Span<byte> change as it wouldn't play nice with the async function that exports whole folders.

What's the problem?

Reply:
I tried to make ContentFile a ref struct (to hold Span) but it tells me that is stack based and cant be used in async functions.